src/base/utils/fs.h

```diff
@@ -67,6 +67,40 @@ namespace Utils
         /* End of Qt4 code */
 
         QString cacheLocation();
+
+        /// Utility class to defer file deletion
+        class FileProcessingGuard
+        {
+        public:
+            FileProcessingGuard(const QString &path = QString());
+            ~FileProcessingGuard();
+
+            /// Cancels deferred file deletion
+            void releaseFile();
+
+        private:
+            QString m_path;
+        };
+
+        /// reads settings for .torrent files from preferences
+        /// and sets up file guard accordantly
+        class TorrentFileProcessignGuard : private FileProcessingGuard
```

Comment:
I agree with you partially. But...
1. Objective correspondence between settings names and names of the components/properties is better than subjective separation into groups.
2. qBittorrent configuration file is mostly implementation detail and it isn't a user configuration interface.